tests/phpunit/elementor/includes/test-utils.php

```diff
@@ -180,4 +180,141 @@ public function test_is_empty() {
 		$this->assertEquals( false, Utils::is_empty( [ 'key' => '0' ], 'key' ),
 			"[ 'key' => '0' ] is empty" );
 	}
+
+	public function test_get_current_domain() {
+		// Scenario 1 - Website without a root folder.
+		$this->get_current_domain__base();
+
+		// Scenario 2 - Website with a root folder.
+		$this->get_current_domain__root_folder();
+	}
+
+	public function get_current_domain__base() {
+		$initial_server_name = $_SERVER['SERVER_NAME'];
+
+		// Overwriting the SERVER_NAME to emulate a subdomain scenario.
+		$_SERVER['SERVER_NAME'] = 'www.my-website.com';
+
+		$initial_php_self = $_SERVER['PHP_SELF'];
+
+		// Setting a root folder to the website.
+		$_SERVER['PHP_SELF'] = '/index.php';
+
+		$this->assertEquals(
+			'www.my-website.com',
+			Utils::get_current_domain(),
+			'Current domain value is incorrect.'
+		);
+
+		// Restoring root folder.
+		$_SERVER['PHP_SELF'] = $initial_php_self;
+
+		// Restoring SERVER_NAME value.
+		$_SERVER['SERVER_NAME'] = $initial_server_name;
+	}
+
+	public function get_current_domain__root_folder() {
+		$initial_server_name = $_SERVER['SERVER_NAME'];
+
+		// Overwriting the SERVER_NAME to emulate a subdomain scenario.
+		$_SERVER['SERVER_NAME'] = 'www.my-website.com';
+
+		$initial_php_self = $_SERVER['PHP_SELF'];
+
+		// Setting a root folder to the website.
+		$_SERVER['PHP_SELF'] = '/my-website-folder/index.php';
+
+		$this->assertEquals(
+			'www.my-website.com/my-website-folder',
+			Utils::get_current_domain(),
+			'Current domain value is incorrect.'
+		);
+
+		// Restoring root folder.
+		$_SERVER['PHP_SELF'] = $initial_php_self;
+
+		// Restoring SERVER_NAME value.
+		$_SERVER['SERVER_NAME'] = $initial_server_name;
+	}
+
+	public function test_get_current_domain_src() {
+		// Scenario 1 - Subdomain with no root folder ( www.x.com === de.x.com ).
+		$this->get_current_domain_src__sub_domain();
+
+		// Scenario 2 - Subdomain with a root folder ( www.x.com/site === de.x.com/site ).
+		$this->get_current_domain_src__sub_domain_root_folder();
+
+		// Scenario 3 - Root folder + mirror domain points to root folder ( www.x.com/site === www.y.com ).
+		$this->get_current_domain_src__mirror_domain();
+	}
+
+	public function get_current_domain_src__set_site_url( $url = '' ) {
```

Comment:
A non-test method should be private and at the end of the class

Reply:
It must be public to be used in a filter.